src/gui/shell.cpp

```diff
@@ -137,6 +141,10 @@ bool Shell::setGuiFont(const QString& fdesc, bool force, bool updateOption)
 				weight = QFont::Bold;
 			} else if (attr == "l") {
 				weight = QFont::Light;
+			} else if (attr == "sb") {
+				weight = QFont::DemiBold;
+			} else if (attr.length() > 0 && attr.at(0) == 'w') {
+				weight = (attr.right(attr.length()-1)).toInt();
```

Comment:
It might be good to add some user input sanitization like `:h` does above?

```C++
bool ok{ false };
const int fontWeight{ attr.right(attr.length()-1).toInt(&ok) };
if (!ok || fontWeight < 0 || fontWeight > 99) {
	m_nvim->api0()->vim_report_error("Invalid font weight: 0-99");
	return false;
}
weight = fontWeight;
```

Feel free to ignore, not strictly necessary :)